sdk/azidentity/chained_token_credential.go

```diff
@@ -62,36 +64,40 @@ func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.Token
 		token, err = cred.GetToken(ctx, opts)
 		var credErr credentialUnavailableError
 		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
+			errList = append(errList, err.(credentialUnavailableError))
 		} else if err != nil {
 			var authFailed AuthenticationFailedError
 			if errors.As(err, &authFailed) {
-				err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
+				err = fmt.Errorf("ERROR in GetToken() call for %s:%s\n\t%s", c.name, createChainedErrorMessage(errList), err)
```

Comment:
For what it's worth I prefer "authentication failed" because it's clear and to the point in the very common case of an applications which doesn't call `GetToken()` directly

Reply:
Can do!